src/main/java/org/zalando/nakadi/domain/PaginationLinks.java

```diff
@@ -0,0 +1,46 @@
+package org.zalando.nakadi.domain;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import javax.annotation.concurrent.Immutable;
+import java.util.Optional;
+
+@Immutable
+@JsonInclude(JsonInclude.Include.NON_ABSENT)
+public class PaginationLinks {
+
+    private final Optional<Link> prev;
+
+    private final Optional<Link> next;
+
+    public PaginationLinks() {
+        this(Optional.empty(), Optional.empty());
+    }
+
+    public PaginationLinks(final Optional<Link> prev, final Optional<Link> next) {
```

Comment:
As we decided that we prefer Optionals over nulls I would choose to keep the current implementation.

> because right now we can pass null in constructor.

No, we can't, because the parameters are not marked as `@Nullable`